packages/@uppy/provider-views/src/Item/components/ListLi.js

```diff
@@ -1,65 +1,78 @@
 const { h } = require('preact')
 
-const getAriaLabelOfCheckbox = (props) => {
-  if (props.type === 'folder') {
-    if (props.isChecked) {
-      return props.i18n('unselectAllFilesFromFolderNamed', { name: props.title })
-    }
-    return props.i18n('selectAllFilesFromFolderNamed', { name: props.title })
-  }
-  if (props.isChecked) {
-    return props.i18n('unselectFileNamed', { name: props.title })
-  }
-  return props.i18n('selectFileNamed', { name: props.title })
-}
-
 // if folder:
 //   + checkbox (selects all files from folder)
 //   + folder name (opens folder)
 // if file:
 //   + checkbox (selects file)
 //   + file name (selects file)
-module.exports = (props) => {
+
+function ListItem (props) {
+  const {
+    className,
+    isDisabled,
+    restrictionReason,
+    isCheckboxDisabled,
+    isChecked,
+    toggleCheckbox,
+    type,
+    id,
+    itemIconEl,
+    title,
+    handleFolderClick,
+    showTitles,
+    i18n,
+  } = props
+
   return (
-    <li className={props.className} title={props.isDisabled ? props.restrictionReason : null}>
-      {!props.isCheckboxDisabled ? (
-        <button
-          type="button"
-          className={`uppy-u-reset uppy-ProviderBrowserItem-fakeCheckbox ${props.isChecked ? 'uppy-ProviderBrowserItem-fakeCheckbox--is-checked' : ''}`}
-          onClick={props.toggleCheckbox}
+    <li
+      className={className}
+      title={isDisabled ? restrictionReason : null}
+    >
+      {!isCheckboxDisabled ? (
+        <input
+          type="checkbox"
+          className={`uppy-u-reset uppy-ProviderBrowserItem-fakeCheckbox ${isChecked ? 'uppy-ProviderBrowserItem-fakeCheckbox--is-checked' : ''}`}
+          onChange={toggleCheckbox}
           // for the <label/>
-          id={props.id}
-          role="option"
-          aria-label={getAriaLabelOfCheckbox(props)}
-          aria-selected={props.isChecked}
-          aria-disabled={props.isDisabled}
-          disabled={props.isDisabled}
+          name="listitem"
+          id={id}
+          checked={isChecked}
+          aria-label={type === 'file' ? null : i18n('allFilesFromFolderNamed', { name: title })}
+          aria-checked={isChecked}
+          aria-disabled={isDisabled}
```

Comment:
I don't think we need to add `aria-` attribute when using `<input>` element: https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaChecked

> Where possible use an HTML <input> element with type="checkbox" as this element has built in semantics and does not require ARIA attributes.
```suggestion
```

Reply:
This is true, it was still on there from a previous iteration I did without an input element.